uint8 which_area
uint8 which_point
uint8 isnew